src/FakeItEasy/Configuration/RuleBuilder.cs

```diff
@@ -97,10 +93,10 @@ public virtual IAfterCallSpecifiedConfiguration AssignsOutAndRefParametersLazily
 
         public UnorderedCallAssertion MustHaveHappened(Repeated repeatConstraint)
         {
-            Guard.AgainstNull(repeatConstraint, "repeatConstraint");
+            Guard.AgainstNull(repeatConstraint, nameof(repeatConstraint));
 
             this.manager.RemoveRule(this.RuleBeingBuilt);
-            var asserter = this.asserterFactory.Invoke(this.Calls.Cast<IFakeObjectCall>());
+            var asserter = this.asserterFactory.Invoke(this.Calls);

```

Comment:
IIRC, the cast was necessary before C# 4. But since we dropped .NET 3.5 support, it wouldn't make sense to compile with C# 3 anyway, so :+1: